server/src/main/java/org/opensearch/plugins/NetworkPlugin.java

```diff
@@ -107,4 +107,41 @@ default Map<String, Supplier<HttpServerTransport>> getHttpTransports(
     ) {
         return Collections.emptyMap();
     }
+
+    /**
+     * Returns a map of secure {@link Transport} suppliers.
+     * See {@link org.opensearch.common.network.NetworkModule#TRANSPORT_TYPE_KEY} to configure a specific implementation.
+     */
+    default Map<String, Supplier<Transport>> getSecureTransports(
```

Comment:
For 2.x what do you think about making this an overload of the existing method, and adding a deprecation for the version without a SecureTransportSettingsProvider, but in main removing the original version (breaking change for only 3.x)?

IMO this makes a much clearer path forward that doesn't qualify secure vs insecure.

```suggestion
    default Map<String, Supplier<Transport>> getTransports(
```

Reply:
I thought about that and here is a deal: the `getSecureTransport` is only ever invoked when there is `SecureTransportSettingsProvider` available (which at this moment it won't be). If we won't to unify everything under one API, we either make `SecureTransportSettingsProvider` a mandatory or deal with `nullable` / `Optional` parameters all over the place. I thought that having separate APIs is cleaner at the moment (since this is still `@ExperimentalApi` we could change it)